core/controllers/profile.py

```diff
@@ -180,10 +184,10 @@ def post(self):
         can_receive_email_updates = self.payload.get(
             'can_receive_email_updates')
 
-        has_previously_registered = (
-            user_services.has_user_registered_as_editor(self.user_id))
+        has_ever_registered = bool(user_services.get_username(self.user_id))
```

Comment:
Might be better if this were a function in user_services (checking if user was ever registered) versus having ad-hoc logic here.